### DISCUSSION TITLE
Semester 4 Woche 7

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#48] [Student Exchange]
Hallo zusammen,

diesmal posten wir unseren Blog sehr früh, da wir nach unserer kleinen ⏳ Pause aufgrund der Klausuren unsere Produktivität 📈 sehr stark gesteigert haben, um uns in den kommenden Wochen vollkommen auf die 💻 Implementierung fokussieren zu können.

Diese Woche stand die Einrichtung der 🔁 CI Pipeline an. Wie bereits im [Blog "Semester 4 Woche 4"](https://github.com/Christian-2003/CrInGE/discussions/44) erwähnt, ist diese bereits in Git Hub eingerichtet und berichtet via eines 🏷️ Status badge in der [README.md](https://github.com/Christian-2003/CrInGE/blob/master/README.md) den momentanen Stand der Tests.

Die Funktionalität der Pipeline beschränkt sich auf das 🏗️ Bauen und 🧪 Testen der Anwendung.
Hierdurch soll die ⚙️ Operabilität des Systems nach jedem Push gewährleistet werden.
Dies wird mittels `mvn -B compile --file pom.xml` und `mvn test --file pom.xml` auf der neusten windows Version mit Maven getestet.

Um zu verhindern das nicht richtlinienkonformer Code im 🌐 remote Repository sollte das Linting vor dem commit 📍 lokal durchgeführt werden. Idealerweise geschieht dies via Git Hook in der pre-commit Phase.
Da unser 👥 Team allerdings nicht vertraut mit diesen ist und das Reagieren auf das Feedback des Linters nicht trivial erscheint, bleibt dies ein 🙏 Wunsch für die Zukunft.

Des weiteren hat unser Mitarbeiter des 📅 Monats keine Knechtschaft ausgelassen, um an dem Testbericht weiter zu schreiben.
Somit ist die Dokumentation der 🧪 Test nun fast abgeschlossen und ihr könnt euch zentral eine [Übersicht aller Test](https://github.com/Christian-2003/CrInGE/blob/master/docs/Testbericht.md) 👁️ anschauen.

Das wäre dann auch schon alles für diese Woche. Im nächsten Blog werden wir euch weiter über das Projekt informieren.

**Comment by davidlzt** [comment]
Hallo zusammen,

toller Eintrag! Besonders beeindruckend fand ich die detaillierte Erklärung der Testprozesse. 
Ich freue mich schon auf den nächsten Blog und die weiteren Fortschritte.

Eine kleine Frage: 
Habt ihr Pläne, die CI Pipeline in Zukunft noch weiter auszubauen oder zusätzliche Funktionen zu integrieren?

Viele Grüße!

**Comment by Bjoern04** [comment]
Hallo,

der Blogeintrag sieht soweit sehr gut aus.
Es ist gut erklärt was ihr gemacht habt und eure hinzugefügten Funktionalitäten erscheinen sinnvoll.

Habt ihr noch vor euch weiter mit dem Linting und dem nicht richtlinienkonformen Code zu beschäftigen? 
Falls ja, wäre bei erfolgreicher Umsetzung ein Update in einem der folgenden Blogeinträge sicher interessant.

Weiter so!

**Comment by amjadalnakshbandi** [comment]
Hallo zusammen,

Das war ein großartiger Beitrag! Besonders beeindruckt hat mich die detaillierte Erklärung der Testprozesse. 
Ich freue mich schon auf den nächsten Blogbeitrag und die weiteren Fortschritte.
Gute Arbeit, Jungs 👏
